ccmbioinfo/lfs
# CARDS-2434: Make the S3 export configurable

To test:
- build with `mvn clean install -Pdocker`
- in `compose-cluster` run `python3 generate_compose_yaml.py --dev_docker_image  --composum --cards_project cards4heracles --s3_test_container --oak_filesystem` to generate a docker-compose file
- edit the `docker-compose.yml` file and add to the cards environment `REFERENCE_DATE=2023-01-01`
- start with `docker-compose build && docker-compose up`
- at `http://localhost:9001/` login with `minioadmin / minioadmin` and create a bucket named `uhn`
- login into cards and create three participant status forms for three different patients, leave one of them incomplete
- use composum to change the `jcr:lastModified` date for one of the complete forms to yesterday
- open `http://localhost:8080/Subjects.s3push` to trigger a push
- at `http://localhost:9001/` check that only one file was pushed, the one that was complete and modified today
- at `http://localhost:8080/system/console/configMgr` look for the `S3 Export` configuration for heracles and change the export schedule expression to be one minute from now, e.g. `0 23 15 * * ? *`
- wait one minute, check in minio that the form that was modified yesterday was pushed
- preview these files in minio, make sure that they have the bare format, and that date question is a number
- at `http://localhost:8080/system/console/configMgr` add a new configuration for S3 export, name it `new`, set `new` as the bucket name leave the other values to the default default
- at `http://localhost:9001/` create a new bucket called `new`
- open `http://localhost:8080/Subjects.s3push`, check that an error is displayed
- `http://localhost:8080/Subjects.s3push?config=new`
- at `http://localhost:9001/` check that the form modified today was pushed to the new bucket
- close docker and cleanup with `docker-compose rm -vf && ./cleanup.sh`
